### PR TITLE
Change "Saved" indicator

### PR DESCRIPTION
**Pull request by wvds**
Before submitting, please confirm you've
 - [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

Please provide the following information:

**Summary**
Changes the "Saved" indicator to show up next to the section titles, instead of one generic "Saved" indicator.

**Issue link**
#500 

**Additional Notes**
Also includes a small fix for an `property of undefined`-error when the spellcheck is disabled.

### CODE REVIEW

**Comment by yuya-oc** [comment]
https://circleci.com/gh/mattermost/desktop/1056#artifacts

@jasonblais When clicking app options soon after saving the server list, the indicator moves to the edited part before hiding. Is it okay?
[https://gyazo.com/559c81cf3c00d755154b6f304b2279a8](https://gyazo.com/559c81cf3c00d755154b6f304b2279a8)

**Comment by jasonblais** [comment]
@wvds Would it be feasible to keep the `Saved` indicator even if you click elsewhere on the page?

E.g. (from the GIF posted by @yuya-oc above)
 1. Add or edit server information
 2. Enable spell checking

The `Saved` indicator should remain in the server management section until it automatically fades away, even after enabling spell checking.

**Comment by jasonblais** [comment]
Have also pinged @asaadmahmood to take a brief look at the UI.

Overall great work on this @wvds!!

File: src/browser/components/SettingsPage.jsx [hunk 1]
```diff
@@ -426,6 +427,13 @@ const SettingsPage = createReactClass({
       <Row>
         <Col md={12}>
           <h2 style={settingsPage.sectionHeading}>{'App Options'}</h2>
+          <div className='IndicatorContainer'>
+            <AutoSaveIndicator
+              show={this.state.savingType === 'app'}
+              savingState={this.state.savingState}
```
Comment (yuya-oc) [inline src/browser/components/SettingsPage.jsx]: To take @jasonblais suggestion, you would need to manage different `state` parameters for each indicator.

File: src/browser/components/SettingsPage.jsx [hunk 2]
```diff
@@ -470,6 +472,13 @@ const SettingsPage = createReactClass({
               xs={8}
             >
               <h2 style={settingsPage.sectionHeading}>{'Server Management'}</h2>
+              <div className='IndicatorContainer'>
+                <AutoSaveIndicator
+                  show={this.state.savingType === 'server'}
+                  savingState={this.state.savingState}
```
Comment (yuya-oc) [inline src/browser/components/SettingsPage.jsx]: To take @jasonblais suggestion, you would need to manage different `state` parameters for each indicator.

**Comment by jasonblais** [comment]
Quick follow-up question: What font size is the `Saved` indicator text at the moment?

**Comment by yuya-oc** [comment]
It's 14px.

**Comment by jasonblais** [comment]
Hey @wvds, let me know if you have any questions about the above comments?

**Comment by wvds** [comment]
@yuya-oc I've updated the pull request.
@jasonblais Everything is clear, I've updated it :) Not really a lot of spare time lately

**Comment by jasonblais** [comment]
@yuya-oc @wvds 

Just one minor issue I noticed: When you change a setting, "Saving..." should appear for 0.5 seconds after the most recent change before "Saved!" is shown. However, now "Saved!" appears almost instantaneously.

Not sure if @yuya-oc this is something you'd like to help with?

Otherwise works great. Big thanks @wvds!

**Comment by yuya-oc** [comment]
@jasonblais I'll take a look.

File: src/browser/components/SettingsPage.jsx [hunk 3]
```diff
@@ -100,7 +115,7 @@ const SettingsPage = createReactClass({
     if (teams.length === 0) {
       this.setState({showAddTeamForm: true});
     }
-    setImmediate(this.startSaveConfig);
+    setImmediate(this.startSaveConfig, 'server');
```
Comment (MusikPolice) [inline src/browser/components/SettingsPage.jsx]: Should this string be a constant, so that it's not possible for a typo to break this functionality?

**Comment by yuya-oc** [comment]
https://circleci.com/gh/mattermost/desktop/1093#artifacts

I updated the branch because it's a little difficult to tell indicator's animation in detail. Methods are expanded from original implementation to manage multiple state members.

**Comment by jasonblais** [comment]
Sweet, this looks good to merge, thanks @yuya-oc!

**Comment by yuya-oc** [comment]
Thanks all!

**Pull request merged**